### ISSUE TITLE
[Question] How to specify linker options in profile?

### ISSUE THREAD

**Issue by Poupine** [conan-io/conan#9989]
Hi there,

I've started using conan with mingw64 on WSL2 recently and it all compiles without any issues except for one issue:
I'm using the recipes from conan.io (mostly) with conan install --build missing, which creates static libraries. However those static libraries (at least assimp and/or its dependencies) seem to link dynamically to libstdc++/libgcc. For my own code however, I want to compile with -static-libgcc and -static-libstdc++ because I don't want to have to distibute the C/C++ runtime dlls with my binaries. The problem then is that I get double definitions for the runtime libraries because I end up trying to link to both the static and dynamic version at the same time.
My question is then: is there a way I can set my conan profile to always compile packages with -static-libgcc and -static-libstdc++? Or do I need to adapt the recipes to support that? (The latter I'm hoping I can avoid)

Thanks in advance!

**Comment by SpaceIm** [comment]
You can try to add proper link flags to `LDFLAGS` in env var section of your conanfile. conan-center recipes are designed (or at least there is a best effort) to propagate those kind of standard environment variables to the underlying build system of the library built by the recipe.

**Comment by memsharded** [comment]
> You can try to add proper link flags to LDFLAGS in env var section of your conanfile. conan-center recipes are designed (or at least there is a best effort) to propagate those kind of standard environment variables to the underlying build system of the library built by the recipe.

Yes, the problem @Poupine is that there is no general way to inject compilation or link flags into the build of every different build system out there. Not all build systems in all platforms behave the same, or even listen to environment variables values. So it is possible that even if you set such env-var, not all packages will use it. Even for build systems that do listen to those env-vars, it is not unusual that some values will be hardcoded, overwriting the env-var. ConanCenter community does an amazing job trying to provide for this, but it is impossible to have guarantees.

We have plans to try to improve this leveraging the new ```xxxxToolchain`` generators, but at the moment it is just an idea, a bit early to start it.

**Comment by Poupine** [comment]
Thanks for the info guys! I actually found a solution that does not involve injecting linker flags in conan. It turns out that I was simply missing -static (on top of -static-libgcc/-static-libstdc++) and now things are compiling and linking properly.

**Comment by Techno-Fox** [comment]
what was the solution @Poupine?

**Comment by Poupine** [comment]
@Techno-Fox my answer was not really well formulated.
For mingw64 the solution was indeed to inject linker flags options and I did it globally (since all the recipes required that injection) by adding modifying LDFLAGS at the profile level. One thing to note is: BOTH -static and -static-libstdc++ need to be injected.
Now what I meant (and did a poor job at conveying) when I said I found a solution that does not involve injecting flags is that I am no longer using mingw and instead found out that through wsl2, I can call windows native executables, including MSVC compilation toolbox. I ended up going with that solution in the end as it fits my needs better.

Hope that answers your question
